doc/pac/Port Access Control.md

```diff
@@ -0,0 +1,844 @@
+
+- **Single Host and Multiple Authentication Modes**: All clients on the port need to authenticate. The learning mode of the port is always set to CPU_TRAP. Once a client starts the authentication process, the client is no longer unknown to PAC. PAC installs a static FDB entry to mark the client known so that the incoming traffic does not flood the CPU. The entry is installed with discard bits set to prevent client traffic from being forwarded. In effect, the packets are not flooded to the CPU nor forwarded to other ports during the authentication process. When the client is authenticated, the discard bits of the installed FDB entry are reset to allow client traffic. 
+
+  ​
+
+### 2.2.5 VLAN
+1. PAC associates authenticated clients to a VLAN on the port.
+2. If RADIUS assigns a VLAN to a client, the port's configured untagged VLAN membership is reverted and the RADIUS assigned VLAN is used to authorize the client. The RADIUS assigned VLAN is operationally configured as the untagged VLAN of the port.  All incoming untagged client traffic is assigned to this VLAN. Any incoming client's tagged traffic will be allowed or dropped based on if it matches the port's configured untagged VLAN or not.
+3. If RADIUS does not assign a VLAN to a client, the port's configured untagged VLAN is used to authorize the client. The port's untagged VLAN configuration is retained and all incoming untagged client traffic is assigned to this VLAN. Any incoming client's tagged traffic will be allowed or dropped based on if it matches the port's configured untagged VLAN or not.
+4. All clients on a port are always associated with a single VLAN.
+5. The RADIUS assigned VLAN configured is reverted to the port's configured untagged VLAN once the last authenticated client on the port logs off.
+6. When PAC is disabled on the port, the operationally added untagged VLAN, if present, is removed from the port and the user configured untagged VLAN is assigned back to the port.
+7. If clients are authorized on the port's configured untagged VLAN and the VLAN configuration is modified, all the authenticated clients on the port are removed.
+8. If clients are authorized on RADIUS assigned VLAN, any updates on the port's configured untagged VLAN does not affect the clients. The configuration is updated in the CONFIG_DB but not propagated to the port.
+
+
+### 2.2.6 Warmboot
+
+After a Warm Boot, the authenticated client sessions are torn down and they need to authenticate again.
+
+# 3 Design
+
+## 3.1 Overview
+
+[Figure 2](#configuration-flow) shows the high level design overview of PAC services in SONiC. The existing "macsec" docker is leveraged. 
```

Comment:
why are we re-using macsec docker? why cannot use build a new pac docker?

Reply:
MACsec, 802.1x, PAC and MAB are all related features.
Today Switch-to-Switch MACsec is implemented in the macsec docker.
It may be logically extended to Host-to-Switch MACsec which uses 802.1x to authenticate clients and then derive keys needed for traffic encryption between the client and the switch.
PAC is the central client manager which also uses MAB to authenticate 802.1x unaware clients.
Hence it makes sense to club all these features in the existing MACsec docker.